# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/WebRenderAPI.cpp
# Commit: 1764701d11d1
# Full Hash: 1764701d11d103a2ac02f8cfa5ff4511b8f5dd70
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-22 16:44:08
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Always send parent commands when sending mDestroyedActors r=kats,sotaro
#   
#   If we try to send them separately as we were before, we can run into
#   cases where we try to destroy the actors and then send the OpRemoveTexture,
#   which crashes.
# ==============================================================================

diff -r 32f7793dfd1a -r 1764701d11d1 gfx/webrender_bindings/WebRenderAPI.cpp
--- a/gfx/webrender_bindings/WebRenderAPI.cpp	Thu Mar 21 23:15:32 2019 +0000
+++ b/gfx/webrender_bindings/WebRenderAPI.cpp	Thu Mar 21 23:15:42 2019 +0000
@@ -730,8 +730,8 @@
   wr_api_finalize_builder(SubBuilder(aOutTransaction.mRenderRoot).mWrState,
                           &aOutTransaction.mContentSize,
                           &aOutTransaction.mDLDesc, &dl.inner);
-  aOutTransaction.mDL =
-      ipc::ByteBuf(dl.inner.data, dl.inner.length, dl.inner.capacity);
+  aOutTransaction.mDL.emplace(dl.inner.data, dl.inner.length,
+                              dl.inner.capacity);
   dl.inner.capacity = 0;
   dl.inner.data = nullptr;
 }
